warp10/src/main/java/io/warp10/script/ScriptRunner.java

```diff
@@ -390,16 +390,16 @@ public int compare(String o1, String o2) {
           if (runAtStartup) {
             runnables.add(script);
           } else {
-            Long period = scriptAndPeriod.getValue();
-            long schedat = System.currentTimeMillis();
+            Long period = 1000000L * scriptAndPeriod.getValue();
+            long schedat = System.nanoTime();
 
             if (0 != schedat % period) {
               schedat = schedat - (schedat % period) + period;
             }
```

Comment:
If I'm not mistaken, this is to align with epoch+N*period. As the origin of nanoTime() is unknown, this is not possible to do that like this.